### ISSUE TITLE
Error: Error create profile Ssl (/partition/profile_5601): HTTP 400 :: {"code":400,"message":"01b4002a:3: Client SSL profile (/partition/profile_5601):cert-key-chain and profile cert/key/chain/passphrase options cannot be specified together.","errorStack":[],"apiError":3}

### ISSUE THREAD

**Issue by howsTricks** [F5Networks/terraform-provider-bigip#449]
### Environment

 * TMOS/Bigip Version: 15.x
 * Terraform Version: 0.14.7
 * Terraform bigip provider Version: 1.7

### Summary

We are revisiting some older configuration that has not been touched in awhile (thinking around terraform0.12 age) to update ssl profile configuration and are getting the following error message on apply (not shown during plan):

Error: Error create profile Ssl (/partition/profile_5601): HTTP 400 :: {"code":400,"message":"01b4002a:3: Client SSL profile (/partition/profile_5601):cert-key-chain and profile cert/key/chain/passphrase options cannot be specified together.","errorStack":[],"apiError":3}

### Steps To Reproduce

This seems be related to how cert/key/chain configuration has changed at some point, I see in our old code we used the following format:

```
resource "bigip_ltm_profile_client_ssl" "child-ClientSsl" {
  name = "/${var.partition}/client.ssl.${var.name}"
  partition = "***REDACTED***"
  defaults_from = "/***REDACTED***/***REDACTED***"
  authenticate = "always"
  cert_key_chain {
          name       = "***REDACTED***"
          cert       = "***REDACTED***"
          chain      = "***REDACTED***"
          key        = "***REDACTED***"
  }
}
```

But current examples show something like this:

```
resource "bigip_ltm_profile_client_ssl" "child-ClientSsl" {
  name = "/${var.partition}/client.ssl.${var.name}"
  partition = "***REDACTED***"
  defaults_from = "/***REDACTED***/***REDACTED_NEWPROFILE***"
  authenticate = "always"
  
  cert       = "/***REDACTED***/***REDACTED***"
  chain      = "/***REDACTED***/***REDACTED***"
  key        = "/***REDACTED***/***REDACTED***"

 
}
```

Full example of code being applied, adhering to the new standard. Note we've also tried with the old standard.

```
Error: Error create profile Ssl (/***REDACTED***/***REDACTED***_5601): HTTP 400 :: {"code":400,"message":"01b4002a:3: Client SSL profile (/partition/profile_5601):cert-key-chain and profile cert/key/chain/passphrase options cannot be specified together.","errorStack":[],"apiError":3}




terraform {
  backend "consul" {}
}

provider "bigip" {
    address = "***REDACTED***"
    username = "***REDACTED***"
    password = var.bigip_password
}
   

resource "bigip_ltm_virtual_server" "vip" {
  depends_on = [bigip_ltm_pool.pool,bigip_ltm_profile_client_ssl.child-ClientSsl]
  name = "/${var.partition}/vip.${var.name}"
  destination = "***REDACTED***"
  port = ***REDACTED***
  description = "***REDACTED***"
  pool = "/${var.partition}/pool.${var.name}"
  ip_protocol = "tcp"
  source_address_translation = "automap"
  client_profiles = ["/${var.partition}/client.ssl.${var.name}"]
  profiles = ["/Common/tcp","/Common/http"]
  persistence_profiles = ["/Common/cookie"]
  server_profiles = ["/Common/serverssl"]
}

resource "bigip_ltm_profile_client_ssl" "child-ClientSsl" {
  name = "/${var.partition}/client.ssl.${var.name}"
  partition = "***REDACTED***"
  defaults_from = "/***REDACTED***/***REDACTED_NEWPROFILE***"
  authenticate = "always"
  
  
  cert       = "/***REDACTED***/***REDACTED***"
  chain      = "/***REDACTED***/***REDACTED***"
  key        = "/***REDACTED***/***REDACTED***"

 
}

resource "bigip_ltm_pool" "pool" {
    depends_on = [bigip_ltm_monitor.monitor]
    name = "/${var.partition}/pool.${var.name}"
    monitors = ["/${var.partition}/monitor.${var.name}"]
    allow_nat = "yes"
    allow_snat = "yes"
    load_balancing_mode = "round-robin"
    slow_ramp_time = "10"

}
resource "bigip_ltm_monitor" "monitor" {
  name        = "/${var.partition}/monitor.${var.name}"
  parent      = "/Common/https"
  timeout     = "16"
  interval    = "5"
  send        = "GET /\\r\\n"
}
resource "bigip_ltm_pool_attachment" "node-pool-attach1" {
  depends_on = [bigip_ltm_virtual_server.vip,bigip_ltm_pool.pool]
  pool = "/${var.partition}/pool.${var.name}"
  node = "/${var.partition}/***REDACTED***:5601"
}

resource "bigip_ltm_pool_attachment" "node-pool-attach2" {
  depends_on = [bigip_ltm_virtual_server.vip,bigip_ltm_pool.pool]
  pool = "/${var.partition}/pool.${var.name}"
  node = "/${var.partition}/***REDACTED***:5601"
}
```

### Expected Behavior
SSL profile change to new parent profile (defaults_from)

### Actual Behavior
Error message
Error: Error create profile Ssl (/partition/profile_5601): HTTP 400 :: {"code":400,"message":"01b4002a:3: Client SSL profile (/partition/profile_5601):cert-key-chain and profile cert/key/chain/passphrase options cannot be specified together.","errorStack":[],"apiError":3}

**Comment by nmenant** [comment]
Hi,

Tracking this issue with INFRAANO-290

**Comment by howsTricks** [comment]
Hi @nmenant thanks for the update. To confirm has this been verified? Just wondering if I should try and manually adjust the state to see if we can fix this or if there will be a resolution to this in a near future release? We're wanting to update alot of VIPs with this issue, so want to get going on this either way.

**Comment by howsTricks** [comment]
We were able to build a manual workaround by perfoming the following steps:
1. Configure the bigip_ltm_profile_client_ssl resource to not use cert_key_chain, just cert, key and chain variables like:

```
resource "bigip_ltm_profile_client_ssl" "child-ClientSsl" {
  name = "/${var.partition}/client.ssl.${var.name}"
  partition = "***REDACTED***"
  defaults_from = "/***REDACTED***/***REDACTED_NEWPROFILE***"
  authenticate = "always"
  
  
  cert       = "/***REDACTED***/***REDACTED***"
  chain      = "/***REDACTED***/***REDACTED***"
  key        = "/***REDACTED***/***REDACTED***"

 
}
```

2. Update the remote state to set cert_key_chain to null
`            "cert_key_chain": null,`

3. Run apply and configuration is updated as expected.

**Comment by RavinderReddyF5** [comment]
@howsTricks fixed in v1.10.0, please check and open new issue if u see any problem.
